### ISSUE TITLE
Issues working with remix and any other fetch polyfills or CJS/ESM mismatches

### ISSUE THREAD

**Issue by acoreyj** [panva/oauth4webapi#48]
### What happened?

Using a more browser compliant fetch api like @remix-run/node uses will cause a lot of "response" must be an instance of Response errors due.

Similar to https://github.com/remix-run/react-router/pull/9690

### Version

2.0.6

### Runtime

Node.js

### Runtime Details

Node and Remix framework

### Code to reproduce

```js
import * as o from "oauth4webapi"
    
let discoveryResponse = await o.discoveryRequest(issuer)
const as = await o.processDiscoveryResponse(issuer, discoveryResponse)
```

### Required

- [X] I have searched the issues tracker and discussions for similar topics and couldn't find anything related.
- [X] I agree to follow this project's [Code of Conduct](https://github.com/panva/oauth4webapi/blob/main/CODE_OF_CONDUCT.md)

**Comment by acoreyj** [comment]
The fix would be the same as react-router, implement an isResponse function instead of the instance of. E.g

```
function isResponse(value: any): value is Response {
  return (
    value != null &&
    typeof value.status === "number" &&
    typeof value.statusText === "string" &&
    typeof value.headers === "object" &&
    typeof value.body !== "undefined"
  );
}

```

```
  if (!isResponse(response)) {
    throw new TypeError('"response" must be an instance of Response')
  }
```

If I make a PR is that something you would be good with?

**Comment by panva** [comment]
The issue is that @remix-run/node is not consistent in its globals. I suspected node's own fetch implementation may be meddling in but that's not the case, I checked even with `NODE_OPTIONS='--no-experimental-fetch'`.

The library uses a global fetch() and expects an instanceof global Response to be true on its returned responses. That is a fair expectation of any javascript runtime that has these globals available.

I will not do workarounds for inconsistent runtimes, it starts with Response, then Headers, then CryptoKey.

https://github.com/remix-run/remix/issues/4395 is likely the same issue. The authors of remix seem to think we need to require fetch, but global is a global for a reason.
